packages/react-ui/src/app/features/builder/test-step/step-test-output-hooks.ts

```diff
@@ -0,0 +1,72 @@
+import { flagsHooks } from '@/app/common/hooks/flags-hooks';
+import { Action, FlagId, isEmpty, isNil, Trigger } from '@openops/shared';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { UseFormReturn, useWatch } from 'react-hook-form';
+import { flowsApi } from '../../flows/lib/flows-api';
+
+type FallbackDataInput = (() => unknown) | unknown;
+
+export const stepTestOutputHooks = {
+  useStepTestOutput(
+    flowVersionId: string,
+    stepId: string | undefined,
+    fallbackDataInput: FallbackDataInput,
+  ) {
+    const { data: useNewExternalTestData = false } = flagsHooks.useFlag(
+      FlagId.USE_NEW_EXTERNAL_TESTDATA,
+    );
+
+    const resolveFallbackData = () =>
+      typeof fallbackDataInput === 'function'
+        ? (fallbackDataInput as () => unknown)()
+        : fallbackDataInput;
+
+    return useQuery({
+      queryKey: ['actionTestOutput', flowVersionId, stepId],
+      queryFn: async () => {
+        if (!stepId || !useNewExternalTestData) {
+          return resolveFallbackData();
+        }
+
+        const stepTestOutput = await flowsApi.getStepTestOutput(
+          flowVersionId,
+          stepId,
+        );
+
+        if (isNil(stepTestOutput) || isEmpty(stepTestOutput)) {
+          return resolveFallbackData();
+        }
+
+        return stepTestOutput;
+      },
+    });
+  },
+  useStepTestOutputFormData(
+    flowVersionId: string,
+    form: UseFormReturn<Action> | UseFormReturn<Trigger>,
+  ) {
+    const fallbackData = useWatch({
+      name: 'settings.inputUiInfo.currentSelectedData',
+      control: form.control as any,
+    }) as unknown;
```

Comment:
Can't we just pass a `getFallbackData` to the hook so we don't have the form dependency here?